### PR TITLE
Compatibility before and after RMW pub/sub option structures

### PR DESCRIPTION
**Pull request by rotu**
Includes changes in #46, in a way that makes it work both before and after https://github.com/ros2/rmw/pull/187

### CODE REVIEW

File: rmw_cyclonedds_cpp/CMakeLists.txt [hunk 1]
```diff
@@ -72,6 +72,18 @@ ament_target_dependencies(rmw_cyclonedds_cpp
 
 configure_rmw_library(rmw_cyclonedds_cpp)
 
+string(REGEX REPLACE "([0-9]*)\\.([0-9]*)\\.([0-9]*)((|-).*)" "\\1;\\2;\\3;\\5" parsed_rmw_version "${rmw_VERSION}")
+list(GET parsed_rmw_version 0 rmw_version_major)
+list(GET parsed_rmw_version 1 rmw_version_minor)
+list(GET parsed_rmw_version 2 rmw_version_patch)
```
Comment (dirk-thomas) [inline rmw_cyclonedds_cpp/CMakeLists.txt]: No need to parse those manually. CMake already sets `MAJOR` / `MINOR` / `PATCH` variables based on the `rmw_VERSION` variable.

Comment (rotu) [reply rmw_cyclonedds_cpp/CMakeLists.txt]: Very good call!

**Comment by dirk-thomas** [comment]
The functional changes look good to me. Leaving it up to someone else to merge.

Why are the non-functional changes (line wrapping etc.) part of this PR?

**Comment by eboasson** [comment]
Looks good to me, too. Just verified it on current Eloquent on macOS as well as building against an old binary install of dashing. I'll merge if @dirk-thomas is happy with it too.

Many thanks @rotu for helping out with the cyclone RMW and to @dirk-thomas for getting the RMW versioning in so quickly!

**Comment by rotu** [comment]
> Why are the non-functional changes (line wrapping etc.) part of this PR?

I reformatted the code local to my changes to comply with the C++ developer guide. This should not be controversial; please don't bikeshed.

**Comment by dirk-thomas** [comment]
> This should not be controversial; please don't bikeshed.

The change itself is not controversial. I would just suggest to use a separate PR for unrelated changes - which I don't think is bikeshedding but a commonly agreed upon rational for making pull requests. Anyway - I am not maintaining this repo - so it is just an outsider comment. No need to do anything about it.

**Pull request merged**